### PR TITLE
[BACKLOG-39586] java17 updates for the sub module libraries under reporting

### PR DESCRIPTION
**Pull request by srallapa**
…orting

### CODE REVIEW

File: engine/core/pom.xml [hunk 1]
```diff
@@ -232,6 +226,11 @@
       <artifactId>contiperf</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.github.javatlacati</groupId>
+      <artifactId>contiperf</artifactId>
```
Comment (peterrinehart) [inline engine/core/pom.xml:231]: What is this and why is it needed? Should this have a scope of test?

File: engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/RotationTest.java [hunk 2]
```diff
@@ -109,7 +111,7 @@ public void testXLS() throws ResourceException, IOException {
     }
   }
 
-  @Test
+  //@Test
```
Comment (peterrinehart) [inline engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/RotationTest.java:114]: Let's write a jira ticket to fix these so they don't get forgotten.

File: engine/extensions-drilldown/src/test/java/org/pentaho/reporting/engine/classic/extensions/drilldown/PatternLinkCustomizerTest.java [hunk 3]
```diff
@@ -12,35 +12,30 @@
  * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  * See the GNU Lesser General Public License for more details.
  *
- * Copyright (c) 2019 Hitachi Vantara..  All rights reserved.
+ * Copyright (c) 2024 Hitachi Vantara..  All rights reserved.
```
Comment (peterrinehart) [inline engine/extensions-drilldown/src/test/java/org/pentaho/reporting/engine/classic/extensions/drilldown/PatternLinkCustomizerTest.java:15]: Should be 2019 - 2024

File: engine/extensions/src/test/java/org/pentaho/reporting/engine/classic/extensions/modules/java14print/Java14PrintUtilTest.java [hunk 4]
```diff
@@ -45,7 +45,7 @@
 
 import static org.hamcrest.CoreMatchers.*;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.any;
 import static org.mockito.Mockito.*;
```
Comment (peterrinehart) [inline engine/extensions/src/test/java/org/pentaho/reporting/engine/classic/extensions/modules/java14print/Java14PrintUtilTest.java:49]: Let's expand that to only the classes we're using.

File: libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/JCRSolutionFileModelTest.java [hunk 5]
```diff
@@ -29,8 +29,8 @@
 import static org.hamcrest.CoreMatchers.instanceOf;
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.*;
```
Comment (peterrinehart) [inline libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/JCRSolutionFileModelTest.java:31]: Expand this to just the classes being used.

File: libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/RepositoryFileTreeDtoProxyTest.java [hunk 6]
```diff
@@ -32,7 +32,7 @@
 import static org.hamcrest.CoreMatchers.instanceOf;
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.*;
```
Comment (peterrinehart) [inline libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/RepositoryFileTreeDtoProxyTest.java:34]: Expand the .*

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: engine/core/pom.xml (hunk 1 above)
Comment (srallapa) [reply engine/core/pom.xml:231]: These were eralier refered from "org.databene.contiperf" but with the Unit leatest version updates, it was throwing issues like "java.lang.RuntimeException: java.lang.NoSuchFieldException: fNext" and had to chege the dependency to this with the updated fix compatible with newer junit version.

File: engine/extensions-drilldown/src/test/java/org/pentaho/reporting/engine/classic/extensions/drilldown/PatternLinkCustomizerTest.java (hunk 3 above)
Comment (srallapa) [reply engine/extensions-drilldown/src/test/java/org/pentaho/reporting/engine/classic/extensions/drilldown/PatternLinkCustomizerTest.java:15]: updated.

File: libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/JCRSolutionFileModelTest.java (hunk 5 above)
Comment (srallapa) [reply libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/JCRSolutionFileModelTest.java:31]: expanded.

File: libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/RepositoryFileTreeDtoProxyTest.java (hunk 6 above)
Comment (srallapa) [reply libraries/libpensol/src/test/java/org/pentaho/reporting/libraries/pensol/RepositoryFileTreeDtoProxyTest.java:34]: expanded

File: engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/RotationTest.java (hunk 2 above)
Comment (srallapa) [reply engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/RotationTest.java:114]: https://hv-eng.atlassian.net/browse/BACKLOG-40351 -- This is the ticket I have created to track these.

File: engine/extensions/src/test/java/org/pentaho/reporting/engine/classic/extensions/modules/java14print/Java14PrintUtilTest.java (hunk 4 above)
Comment (srallapa) [reply engine/extensions/src/test/java/org/pentaho/reporting/engine/classic/extensions/modules/java14print/Java14PrintUtilTest.java:49]: expanded

**Pull request merged**